scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -786,4 +870,33 @@ class FossId internal constructor(
             mapOf(SCAN_CODE_KEY to scanCode, SCAN_ID_KEY to scanId, SERVER_URL_KEY to config.serverUrl)
         )
     }
+
+    /**
+     * Return the [PurlType] as determined from the given [url], or null if there is no match. An issue will be added to
+     * [issues] in this case.
+     */
+    private fun urlToPackageType(url: String, issues: MutableList<Issue>): PurlType? =
+        when (val provider = PackageProvider.get(url)) {
+            PackageProvider.COCOAPODS -> PurlType.COCOAPODS
+            PackageProvider.CRATES_IO -> PurlType.CARGO
+            PackageProvider.DEBIAN -> PurlType.DEBIAN
+            PackageProvider.GITHUB -> PurlType.GITHUB
+            PackageProvider.GITLAB -> PurlType.GITLAB
+            PackageProvider.GOLANG -> PurlType.GOLANG
+            PackageProvider.MAVEN_CENTRAL, PackageProvider.MAVEN_GOOGLE -> PurlType.MAVEN
+            PackageProvider.NPM_JS -> PurlType.NPM
+            PackageProvider.NUGET -> PurlType.NUGET
+            PackageProvider.PACKAGIST -> PurlType.COMPOSER
+            PackageProvider.PYPI -> PurlType.PYPI
+            PackageProvider.RUBYGEMS -> PurlType.GEM
+            null -> null
```

Comment:
Can't we just drop this and rely on the `else` below?

Reply:
I added that because the tests were broken. As they use pseudo URLs, `PackageProvider.get` returned `null` and an issue was filled.
However I think filling an issue when the URL cannot be mapped is good practice so I will fix the tests instead.